src/common/pf_eth.c

```diff
@@ -122,6 +149,18 @@ int pf_eth_recv (pnal_eth_handle_t * eth_handle, void * arg, pnal_buf_t * p_buf)
    uint16_t ix = 0;
    pnet_t * net = (pnet_t *)arg;
 
+#ifndef UNIT_TEST
+   /* Reject unicast packets with destination not matching
```

Comment:
Wouldn't it be better if unicast filtering was done at a lower level (ie pnal)? Documentation could be added to pnal_eth_init() to require that incoming frames with unknown unicast destination MAC address should be dropped.

Reply:
I agree, we will start with only stop using promiscuous mode and retest to verify that this solves the original problem.